### PR TITLE
Bytecode comparison on CircleCI

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

**Comment by ekpyron** [comment]
Well, that's annoying...
Apparently CircleCI doesn't let us attach workspaces containing conflicting files to the same job.
But it also doesn't allow us to specify which files to attach and workspaces are "append only".

So far the only way to get around this would be to use separately named build directories for all jobs and persist with them in the path. Which is, well, annoying...

**Comment by ekpyron** [comment]
For reference: https://circleci.com/blog/deep-diving-into-circleci-workspaces/
From what I can see they really provide no means around this whatsoever.
So what I'm doing now is just moving the osx workspace files into the subdirectory ``build/``, while for the others remain as they were. That's hacky, but the quickest solution I could come up with.

**Comment by ekpyron** [comment]
Ok, we do have an discrepancy, but probably not a serious one. The version numbers are:

ems: 0.7.4-develop.2020.10.14+commit.d2eef326
win: 0.7.4-develop.2020.10.14+commit.d2eef326
osx: 0.7.4-nightly.2020.10.14+commit.d2eef326
ubu: 0.7.4-nightly.2020.10.14+commit.d2eef326

So half of them call themselves nightly, the other half develop builds, apparently. I'll figure out where this is set...

**Comment by ekpyron** [comment]
Ok - since we only actually build emscripten nightlies, I see no reason why the CI script should have its builds call themselves nightlies. Let's see if that does it.

**Comment by ekpyron** [comment]
I also tried to have the ``b_win`` build produce a release binary, when built on tags - otherwise the bytecode comparison would start breaking on releases :-). But we'll probably just need to see if that's enough at the next release.

**Comment by ekpyron** [comment]
So... what about merging this like it is, s.t. we do have bytecode comparisons again, and then let @aarlt and @cameel figure out how to do this more nicely in the next two weeks ;-)? Un-drafting this.

File: .circleci/config.yml [hunk 1]
```diff
@@ -592,7 +592,12 @@ jobs:
       - run: *run_build
       - store_artifacts: *artifacts_solc
       - store_artifacts: *artifacts_tools
-      - persist_to_workspace: *artifacts_build_dir
+      - persist_to_workspace:
+          root: .
+          paths:
+            - build/solc/solc
+            - build/test/soltest
+            - build/test/tools/solfuzzer
```
Comment (ekpyron) [inline .circleci/config.yml:600]: This is the main workaround. osx moves these files in a subdirectory ``build``, while the rest has one directory level less.

File: scripts/ci/build.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ ROOTDIR="$(dirname "$0")/../.."
 cd "${ROOTDIR}"
 
 # shellcheck disable=SC2166
-if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]; then echo -n >prerelease.txt; else date -u +"nightly.%Y.%-m.%-d" >prerelease.txt; fi
```
Comment (chriseth) [inline scripts/ci/build.sh:8]: Where do we make sure that this is called "nightly" for the actual emscripten build?

Comment (ekpyron) [reply scripts/ci/build.sh:8]: Those are built in solc-bin from this github action https://github.com/ethereum/solc-bin/blob/gh-pages/.github/workflows/nightly-emscripten.yml

Comment (ekpyron) [reply scripts/ci/build.sh:8]: Hm... which just calls back ``scripts/travis-emscripten/build_emscripten.sh`` ultimately...

Comment (ekpyron) [reply scripts/ci/build.sh:8]: Maybe we actually don't :-) - I'll have a look and make sure - in any case, they won't involve this script, so they're unaffected by the change here.

Comment (ekpyron) [reply scripts/ci/build.sh:8]: Yeah, we don't - they are called ``develop``... but that needs to be fixed in solc-bin. I'll create a PR there.

**Review by chriseth — APPROVED** [review]

**Pull request merged**